crates/spawner/src/draft.rs

```diff
@@ -126,3 +142,69 @@ fn update_draft(
         }
     }
 }
+
+/// Materials for the invalid and valid placing states
+#[derive(Clone, Resource)]
+struct DraftMaterials {
+    valid_placement: Handle<StandardMaterial>,
+    invalid_placement: Handle<StandardMaterial>,
+}
+
+fn insert_materials(mut commands: Commands, mut materials: ResMut<Assets<StandardMaterial>>) {
+    commands.insert_resource(DraftMaterials {
+        valid_placement: materials.add(Color::rgba(0.2, 0.8, 0.2, 0.7).into()),
+        invalid_placement: materials.add(Color::rgba(0.86, 0.08, 0.24, 0.7).into()),
+    });
+}
+
+// Assign the appropriate allowed to all entities in the spawned glb scene
+fn update_object_material(
+    allowed: bool,
+    entities: impl Iterator<Item = Entity>,
+    standard_materials: &mut Query<&mut Handle<StandardMaterial>>,
+    draft_materials: &Res<DraftMaterials>,
+) {
+    for entity in entities {
+        let Ok(mut material_handle) = standard_materials.get_mut(entity) else{
+            continue;
+        };
+        if allowed {
+            *material_handle = draft_materials.valid_placement.clone();
+        } else {
+            *material_handle = draft_materials.invalid_placement.clone();
+        }
+    }
+}
+
+fn update_draft_colour(
+    mut draft_query: Query<(&Draft, &Children, &mut DraftHighlight)>,
+    scene_instances_query: Query<&SceneInstance>,
+    mut standard_materials: Query<&mut Handle<StandardMaterial>>,
+    scene_spawner: Res<SceneSpawner>,
+    draft_materials: Res<DraftMaterials>,
+) {
+    for (draft, children, mut draft_highlight) in &mut draft_query {
+        let allowed = draft.allowed();
+        // Skip assigning material if the allowed state hasn't changed
+        if draft_highlight.previously_allowed == Some(allowed) {
+            continue;
+        }
```

Comment:
I'm struggling to get this working with change detection as for the first few frames after the draft is spawned, the scene doesn't exist.

Reply:
I see. I would create a ticket for scene pre-loading (e.g. during loading stage) and reference the ticket here in a comment. Using change detection would be nicer.

Reply:
I managed to get the change detection working by adding a system that checks with a `SceneInstance` component is added.

Reply:
Cool, I like the current solution!